SMLHelper/Patchers/PrefabDatabasePatcher.cs

```diff
@@ -42,18 +53,26 @@ internal static bool GetPrefabAsync_Prefix(ref IPrefabRequest __result, string c
             return true;
         }
 
-        internal static void Patch(Harmony harmony)
+        internal static void PrePatch(Harmony harmony)
         {
-            harmony.Patch(AccessTools.Method(typeof(PrefabDatabase), nameof(PrefabDatabase.LoadPrefabDatabase)),
-                postfix: new HarmonyMethod(AccessTools.Method(typeof(PrefabDatabasePatcher), nameof(PrefabDatabasePatcher.LoadPrefabDatabase_Postfix))));
-
             harmony.Patch(AccessTools.Method(typeof(PrefabDatabase), nameof(PrefabDatabase.GetPrefabForFilename)), 
                 prefix: new HarmonyMethod(AccessTools.Method(typeof(PrefabDatabasePatcher), nameof(PrefabDatabasePatcher.GetPrefabForFilename_Prefix))));
 
+            harmony.Patch(AccessTools.Method(typeof(PrefabDatabase), nameof(PrefabDatabase.TryGetPrefabFilename)),
+                prefix: new HarmonyMethod(AccessTools.Method(typeof(PrefabDatabasePatcher), nameof(PrefabDatabasePatcher.TryGetPrefabFilename_Prefix))));
+
```

Comment:
For performance, this Harmony pre-patch should be unpatched after `LoadPrefabDatabase_Postfix()` has run.  
All modded prefabs should be in the database once completed.

Reply:
Guess I am still learning lol ..... How exactly do you Unpatch a patch?

Reply:
Ask @tobeyStraitjacket or look at how he unpatches the QModManager Initialize method.
https://github.com/SubnauticaModding/QModManager/blob/master/QMMLoader/QMMLoader.cs#L62

Reply:
> https://github.com/SubnauticaModding/SMLHelper/pull/176/commits/b066fdaf2c028b221a472ccd00de3cf49dea41a0

No no, this isn't right.  
The new `TryGetPrefabFilename_Prefix` should be unpatched as part of the _execution_ of `LoadPrefabDatabase_Postfix`, so that this prefix patch is still there until we can be certain that all mod prefabs have been copied into the prefab database.  
Once copied, then the prefix patch is no longer needed.

Reply:
Fixed as of cd9968f

Reply:
I would say it might also be worth saving a reference to the targets so you don't need to call `AccessTools.Method` multiple times, just for a bit of extra performance, as I don't think `AccessTools` caches the lookups. But it might not really be that much of a difference.